services/visual_recognition/v3.js

```diff
@@ -62,301 +37,329 @@ module.exports = function (RED) {
   RED.httpAdmin.get('/watson-visual-recognition/vcap', function (req, res) {
     res.json(service ? {bound_service: true} : null);
   });
-
-  // Utility functions that check for image buffers, urls and stream data in
  
   function imageCheck(data) {
     return data instanceof Buffer && imageType(data) !== null;
-  };
+  }
 
   function urlCheck(str) {
     var parsed = url.parse(str);
     return (!!parsed.hostname && !!parsed.protocol && str.indexOf(' ') < 0);
-  };
+  }
 
   function stream_buffer(file, contents, cb) {
     fs.writeFile(file, contents, function (err) {
-      if (err) throw err;
+      if (err) {
+        throw err;
+      }
       cb();
     });
-  };
+  }
+
+  function verifyPayload(node, msg) {
+    if (!msg.payload) {
+      this.status({fill:'red', shape:'ring', text:'missing payload'}); 
+      node.error('Missing property: msg.payload', msg);
+      return false;
+    }
+    return true;
+  }
+
+  function verifyInputs(feature, node, msg) {
+    switch(feature) {
+    case 'classifyImage':
+    case 'detectFaces':
+    case 'recognizeText':
+      if (typeof msg.payload === 'boolean' || typeof msg.payload === 'number') {
+        this.status({fill:'red', shape:'ring', text:'bad format payload'}); 
+        node.error('Bad format : msg.payload must be a URL string or a Node.js Buffer', msg);
+        return false;
+      }
+    }
+    return true;
+  }
 
-  // Utility function that performs the Watson Visual Recognition call. 
-  // the cleanup removes the temp storage, and I am not sure whether 
-  // it should be called here or after alchemy returns and passed
-  // control back to cbdone.
+  function verifyServiceCredentials(node, msg) {
+    // If it is present the newly provided user entered key 
+    // takes precedence over the existing one. 
+    node.apikey = sAPIKey || node.credentials.apikey;
+    if (!node.apikey) {
+      node.status({fill:'red', shape:'ring', text:'missing credentials'});          
+      var message ='Missing Watson Visual Recognition API service credentials';
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Infix operators must be spaced.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2737057949/issues/source?bid=3225115&fileBranchId=3352493#l88)